File: src/aikataulu/viikko1/esimerkit/Kahvikuppi.java


class Kahvikuppi {
  
  /* Luokkamuuttujat */
  private String tekstiKyljessa
  private int tayttoaste = 0;

  /* Konstruktori */
  Kahvikuppi(String teksti) { 
    this.tekstiKyljessa = teksti;
  }
  
  /* Metodeja */
  void kaadaKahvia(int maara) {
    tayttoaste += $maara;
    if (tayttoaste > 100) {
      throw new Exception("Kahvikuppi valui yli");
    }
  }
  void juoTyhjaksi() {
    tayttoaste = 0;
  }
  int getKahvinMaara() {
    return tayttoaste;
  }
}

File: src/aikataulu/viikko1/esimerkit/kahvikuppi.php
<?php

class Kahvikuppi {

  /* Luokkamuuttujat */
  private $tekstiKyljessa;
  private $tayttoaste = 0;
  
  /* Konstruktori */
  function __construct($teksti) {
    $this->tekstiKyljessa = $teksti;
  }
  
  /* Metodeja */
  function kaadaKahvia($maara) {
    $this->tayttoaste += $maara;
    if ($this->tayttoaste > 100) {
      throw new Exception("Kahvikuppi valui yli");
    }
  }
  function juoTyhjaksi() {
    $this->tayttoaste = 0;
  }
  function getKahvinMaara() {
    return $this->tayttoaste;
  }
}
